akka-http-caching/src/main/scala/akka/http/caching/LfuCache.scala

```diff
@@ -23,14 +25,14 @@ import scala.compat.java8.FunctionConverters._
 object LfuCache {
 
   def apply[K, V](implicit system: ActorSystem): akka.http.caching.scaladsl.Cache[K, V] =
-    apply(akka.http.caching.scaladsl.CachingSettings(system))
+    apply(scaladsl.CachingSettings(system))
 
   /**
    * Creates a new [[akka.http.caching.LfuCache]], with optional expiration depending
    * on whether a non-zero and finite timeToLive and/or timeToIdle is set or not.
    */
-  def apply[K, V](cachingSettings: CachingSettings): akka.http.caching.scaladsl.Cache[K, V] = {
-    val settings = cachingSettings.lfuCacheSettings.asInstanceOf[LfuCacheSettings]
+  def apply[K, V](cachingSettings: scaladsl.CachingSettings): akka.http.caching.scaladsl.Cache[K, V] = {
```

Comment:
rather than a "leak" this was done so on purpose actually ;)

Reply:
Since scaladsl extends javadsl, scaladsl can take either type. We have this pattern in a number of places. But I'm ok with accepting this PR I think, it's not commonly utilised in practice I guess (that capability)

Reply:
Okay, I was not sure whether this was on  purpose which is why I put it in a separate PR.